### PR TITLE
Decompile/port event_object_movement

### PR DESCRIPTION
**Pull request by ultima-soul**

### CODE REVIEW

File: src/event_object_movement.c [hunk 1]
```diff
@@ -0,0 +1,7436 @@
...
+
+void MoveObjectEventToMapCoords(struct ObjectEvent *objectEvent, s16 x, s16 y)
+{
+    struct Sprite *sprite;
+    const struct ObjectEventGraphicsInfo *graphicsInfo;
+
+    sprite = &gSprites[objectEvent->spriteId];
+    graphicsInfo = GetObjectEventGraphicsInfo(objectEvent->graphicsId);
+    SetObjectEventCoords(objectEvent, x, y);
+    SetSpritePosToMapCoords(objectEvent->currentCoords.x, objectEvent->currentCoords.y, &sprite->pos1.x, &sprit
```
Comment (ProjectRevoTPP) [inline src/event_object_movement.c]: 🤔

Comment (ProjectRevoTPP) [reply src/event_object_movement.c]: 7411

File: src/event_object_movement.c [hunk 2]
```diff
@@ -0,0 +1,8372 @@
...
+    {
+        if (gPlayerReflectionPaletteSets[i].tag == tag)
+        {
+            PatchObjectPalette(gPlayerReflectionPaletteSets[i].data[sCurrentReflectionType], gReflectionEffectPaletteMap[slot]);
+            return;
+        }
+    }
+}
+
+void LoadSpecialObject
```
Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: constant

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: constant

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: constant

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: constant

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: constant

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: constant

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: constant

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: constant

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: `TRUE`

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: `FALSE`

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: `TRUE`

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: `FALSE`

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: `FALSE`

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: `TRUE`

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: `FALSE`. Not going to comment on this anymore, please fix all instances in this file

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: so it doesn't match without the 2 local variables?

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: no space before ++ --

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: align the backslashes

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: constant

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: use `TRUE` `FALSE`

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: wrong indent level for switch

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: no space after `--`

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: use `TRUE` `FALSE`

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: use `TRUE` `FALSE`
not  going to comment on every `hasShadow`

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: Can you update it?

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: are you sure it's not negative value?

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: These casts look really weird. I guess they themselves are s16 (needs fixup elsewhere probably)

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: use `TRUE` `FALSE`

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: use `TRUE` `FALSE`

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: use `TRUE` `FALSE`

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: use `TRUE` `FALSE`

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: use `TRUE` `FALSE`

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: `FALSE`

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: `FALSE`

Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: `FALSE`

File: include/field_control_avatar.h [hunk 3]
```diff
@@ -8,6 +8,8 @@ extern u32 gUnknown_3005078;
 void RestartWildEncounterImmunitySteps(void);
 void ClearPoisonStepCounter(void);
 void SetCableClubWarp(void);
+void sub_806DE28(struct ObjectEvent *);
 void dive_warp(struct MapPosition * pos, u16 behavior);
 
+
```
Comment (PikalaxALT) [inline include/field_control_avatar.h]: Spurious newlines

File: include/fieldmap.h [hunk 4]
```diff
@@ -11,7 +11,7 @@
 #define NUM_PALS_TOTAL 13
 #define VIRTUAL_MAP_SIZE 0x2800
 
-extern struct BackupMapLayout VMap;
+extern struct BackupMapLayout gBackupMapLayout;
```
Comment (PikalaxALT) [inline include/fieldmap.h]: Revert this for assert statements

File: src/event_object_80688E4.c [hunk 5]
```diff
@@ -1,7 +1,6 @@
 #include "global.h"
```
Comment (PikalaxALT) [inline src/event_object_80688E4.c]: Append the contents of this file to event_object_movement.c

File: src/fieldmap.c [hunk 6]
```diff
@@ -7,6 +7,8 @@
 #include "quest_log.h"
 #include "fieldmap.h"
 
+#define VMap gBackupMapLayout // Needed for AGB_ASSERT_EX to match
```
Comment (PikalaxALT) [inline src/fieldmap.c]: Revert name of gBackupMapLayout rather than use this hack

File: data/event_object_movement.s [hunk 7]
```diff
@@ -5,10 +5,10 @@
 gReflectionEffectPaletteMap:: @ 835B934
```
Comment (PikalaxALT) [inline data/event_object_movement.s]: Port to C

File: src/event_object_movement.c (hunk 2 above)
Comment (PikalaxALT) [inline src/event_object_movement.c]: Remove the comment one way or another (event_object_movement.c line 1804)

Comment (PikalaxALT) [inline src/event_object_movement.c]: event_object_movement.c:2700-2701: When porting from data/event_object_movement.s, replace this memcpy with array initialization.

Comment (ultima-soul) [reply src/event_object_movement.c]: These will be uncommented when data is decompiled, which will not be covered in this PR.

Comment (ultima-soul) [reply src/event_object_movement.c]: Tried fiddling with it. No luck.

**Pull request merged**